frame/staking/src/lib.rs

```diff
@@ -1559,8 +1638,13 @@ decl_module! {
 		/// - Independent of the arguments. Insignificant complexity.
 		/// - Contains a limited number of reads.
 		/// - Writes are limited to the `origin` account key.
+		/// ----------
+		/// Base Weight: 25.22 µs
+		/// DB Weight:
+		/// - Read: Bonded, Ledger New Controller, Ledger Old Controller
+		/// - Write: Bonded, Ledger New Controller, Ledger Old Controller
 		/// # </weight>
-		#[weight = 750_000_000]
+		#[weight = 25 * WEIGHT_PER_MICROS + T::DbWeight::get().reads_writes(3, 3)]
```

Comment:
How detailed do we want to be with refunds? For example here we can refund 1 reads and 3 writes if we error on `Err(Error::<T>::AlreadyPaired)?`. Probably has little impact though. Also makes the code very hard to maintain of all errors were to return refund?

Reply:
if we return an err based on user input, there is no need for the moment to do a refund. could be something we optimize in the future, but users should avoid causing errs